x/ccv/types/errors.go

```diff
@@ -6,23 +6,24 @@ import (
-	ErrInvalidVSCMaturedTime    = errorsmod.Register(ModuleName, 12, "invalid maturity time for VSC packet")
-	ErrInvalidConsumerState     = errorsmod.Register(ModuleName, 13, "provider chain has invalid state for consumer chain")
-	ErrInvalidConsumerClient    = errorsmod.Register(ModuleName, 14, "ccv channel is not built on correct client")
-	ErrInvalidProposal          = errorsmod.Register(ModuleName, 15, "invalid proposal")
-	ErrInvalidHandshakeMetadata = errorsmod.Register(ModuleName, 16, "invalid provider handshake metadata")
-	ErrChannelNotFound          = errorsmod.Register(ModuleName, 17, "channel not found")
-	ErrClientNotFound           = errorsmod.Register(ModuleName, 18, "client not found")
-	ErrDuplicateConsumerChain   = errorsmod.Register(ModuleName, 19, "consumer chain already exists")
-	ErrConsumerChainNotFound    = errorsmod.Register(ModuleName, 20, "consumer chain not found")
+	ErrInvalidPacketData           = errorsmod.Register(ModuleName, 2, "invalid CCV packet data")
+	ErrInvalidPacketTimeout        = errorsmod.Register(ModuleName, 3, "invalid packet timeout")
+	ErrInvalidVersion              = errorsmod.Register(ModuleName, 4, "invalid CCV version")
+	ErrInvalidChannelFlow          = errorsmod.Register(ModuleName, 5, "invalid message sent to channel end")
+	ErrInvalidConsumerChain        = errorsmod.Register(ModuleName, 6, "invalid consumer chain")
+	ErrInvalidProviderChain        = errorsmod.Register(ModuleName, 7, "invalid provider chain")
+	ErrInvalidStatus               = errorsmod.Register(ModuleName, 8, "invalid channel status")
+	ErrInvalidGenesis              = errorsmod.Register(ModuleName, 9, "invalid genesis state")
+	ErrDuplicateChannel            = errorsmod.Register(ModuleName, 10, "CCV channel already exists")
+	ErrInvalidVSCMaturedId         = errorsmod.Register(ModuleName, 11, "invalid vscId for VSC packet")
+	ErrInvalidVSCMaturedTime       = errorsmod.Register(ModuleName, 12, "invalid maturity time for VSC packet")
+	ErrInvalidConsumerState        = errorsmod.Register(ModuleName, 13, "provider chain has invalid state for consumer chain")
+	ErrInvalidConsumerClient       = errorsmod.Register(ModuleName, 14, "ccv channel is not built on correct client")
+	ErrInvalidProposal             = errorsmod.Register(ModuleName, 15, "invalid proposal")
+	ErrInvalidHandshakeMetadata    = errorsmod.Register(ModuleName, 16, "invalid provider handshake metadata")
+	ErrChannelNotFound             = errorsmod.Register(ModuleName, 17, "channel not found")
+	ErrClientNotFound              = errorsmod.Register(ModuleName, 18, "client not found")
+	ErrDuplicateConsumerChain      = errorsmod.Register(ModuleName, 19, "consumer chain already exists")
+	ErrConsumerChainNotFound       = errorsmod.Register(ModuleName, 20, "consumer chain not found")
+	ErrInvalidDoubleVotingEvidence = errorsmod.Register(ModuleName, 21, "invalid consumer double voting evidence")
```

Comment:
I'm wondering if it would anything if we had a similar error type for invalid misbehaviour evidence?

Reply:
Because IBC already defines an [`ErrInvalidMisbehaviour`](https://github.com/cometbft/cometbft/blob/v0.37.2/evidence/verify.go#L169).

Reply:
Maybe it would add a bit value having a comment in this `errors.go` file above (or below) `ErrInvalidDoubleVotingEvidence` on this that `ErrInvalidMisbehaviour` pre-exists in IBC.